kernel/kernel-pdfgenerator/src/main/java/io/mosip/kernel/pdfgenerator/impl/PDFGeneratorImpl.java

```diff
@@ -0,0 +1,373 @@
+		}
+		try {
+			Document document = Jsoup.parse(template);
+			document.outputSettings().syntax(Document.OutputSettings.Syntax.xml);
+			PdfRendererBuilder builder = FontPdfRendererBuilder.getBuilder();
+			builder.withHtmlContent(document.html(), null);
+			builder.toStream(os);
+			builder.run();
+		}catch (Exception e) {
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
+					e.getMessage(), e);
+		}
+		return os;
+	}
+
+	@Override
+	public void generate(String templatePath, String outputFilePath, String outputFileName) throws IOException {
+		File outputFile = new File(outputFilePath + outputFileName + OUTPUT_FILE_EXTENSION);
+		try {
+
+
+		try (PDDocument document = new PDDocument()) {
+			PDPage page = new PDPage(PDRectangle.A4);
+			document.addPage(page);
+
+			try (BufferedReader reader = new BufferedReader(new FileReader(templatePath));
+				 PDPageContentStream contentStream = new PDPageContentStream(document, page)) {
+				contentStream.beginText();
+				contentStream.setFont(PDType1Font.HELVETICA, 12);
+				contentStream.newLineAtOffset(50, 750);
+
+				String line;
+				while ((line = reader.readLine()) != null) {
+					contentStream.showText(line);
+					contentStream.newLineAtOffset(0, -15); // Move down for the next line
+				}
+
+				contentStream.endText();
+			}
+
+			document.save(outputFile);
+		}
+		}catch (Exception e){
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
+					e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public OutputStream generate(InputStream is, String resourceLoc) throws IOException {
+		return generate(is); // Placeholder, as no specific resource handling is demonstrated in the existing implementation
+	}
+
+	@Override
+	public byte[] asPDF(List<BufferedImage> bufferedImages) throws IOException {
+		try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+			 PDDocument document = new PDDocument()) {
+
+			for (BufferedImage bufferedImage : bufferedImages) {
+				PDPage page = new PDPage(PDRectangle.A4);
+				document.addPage(page);
+
+				PDImageXObject image = LosslessFactory.createFromImage(document, bufferedImage);
+
+				try (PDPageContentStream contentStream = new PDPageContentStream(document, page)) {
+					contentStream.drawImage(image, 50, 100, 500, 700); // Example scaling
+				}
+			}
+
+			document.save(byteArrayOutputStream);
+			return byteArrayOutputStream.toByteArray();
+		}
+	}
+
+	@Override
+	public byte[] mergePDF(List<URL> pdfFiles) throws IOException {
+		try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+			PDFMergerUtility merger = new PDFMergerUtility();
+			merger.setDestinationStream(byteArrayOutputStream);
+			for (URL url : pdfFiles) {
+				try (PDDocument tempDocument = PDDocument.load(url.openStream())) {
+					merger.addSource(url.openStream());  // Add each file to the merger
```

Comment:
Where is this tempDocument used? As we have already invoked openStream() on the input URL why don't we use the same?

Reply:
modified the method